analyzers/src/SonarAnalyzer.CSharp/Rules/NativeMethodsShouldBeWrapped.cs

```diff
@@ -34,20 +34,28 @@ public sealed class NativeMethodsShouldBeWrapped : SonarDiagnosticAnalyzer
         protected override void Initialize(SonarAnalysisContext context)
         {
             context.RegisterSymbolAction(ReportPublicExternalMethods, SymbolKind.Method);
-            context.RegisterSyntaxNodeAction(ReportTrivialWrappers, SyntaxKind.MethodDeclaration);
+            context.RegisterSyntaxNodeActionInNonGenerated(ReportTrivialWrappers, SyntaxKind.MethodDeclaration);
```

Comment:
[education] why is this necessary to not be raised on generated code?

I get why it's necessary for the case of `LibraryImport` (to not raise the issue twice + users don't control the source generator) - is it the case that in general we should not raise on generated code ever?

Reply:
Yes. It was requested by Pavel at one point that we should fix this here. I added it a item 4 on the todo list here https://github.com/SonarSource/sonar-dotnet/issues/6488 when he suggested it. It is also needed to properly support the LibraryImport attribute.

Reply:
the PR name should reflect that IMO because it's a change of functionality for the whole rule and it's not only related to adding support for a new attribute